torch/fx/passes/shape_prop.py

```diff
@@ -111,9 +111,24 @@ def forward(self, x):
          module (GraphModule): The module to be executed
 
     """
+    def __init__(self, gm, fake_mode=None):
+        super().__init__(gm)
+        if fake_mode:
+            from torch._dynamo.utils import deepcopy_to_fake_tensor
+            self.fake_module = deepcopy_to_fake_tensor(self.module, fake_mode)
+        else:
+            self.fake_module = None
+
+        self.real_module = self.module
```

Comment:
This is perplexing. Why didn't the caller of ShapeProp fakeify the module if they wanted it fake execution.

Reply:
We need fake execution cause the inputs are fake, however, we cannot fakify the module - because we need to write to the `tensor_meta` of the real module. So we fakify to produce a `result` (L130 below), to extract tensor meta, and then keep going. 

If we were to fakify, we would write to the wrong node, and then downstream fusion would be missing the tensor_meta. 

We could also do a full fakify and add a layer to remap the tensor_metas, but that seemed overcomplicated.

Reply:
OK. Worth a comment in that regard

Reply:
Agreed